gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1276 @@
+    if (wrapper == null) {
+      return; // no such zone exists; it might have been deleted by another thread
+    }
+    AtomicReference<ImmutableSortedMap<String, ResourceRecordSet>> dnsRecords =
+        wrapper.dnsRecords();
+    while (true) {
+      // managed zone must have a set of records which is not null
+      ImmutableSortedMap<String, ResourceRecordSet> original = dnsRecords.get();
+      // the copy will be populated when handling deletions
+      SortedMap<String, ResourceRecordSet> copy = new TreeMap<>();
+      // apply deletions first
+      List<ResourceRecordSet> deletions = change.getDeletions();
+      if (deletions != null) {
+        for (Map.Entry<String, ResourceRecordSet> entry : original.entrySet()) {
+          if (!deletions.contains(entry.getValue())) {
+            copy.put(entry.getKey(), entry.getValue());
+          }
+        }
+      } else {
+        copy.putAll(original);
+      }
+      // apply additions
+      List<ResourceRecordSet> additions = change.getAdditions();
+      if (additions != null) {
+        for (ResourceRecordSet addition : additions) {
+          ResourceRecordSet rrset = new ResourceRecordSet();
+          rrset.setName(addition.getName());
+          rrset.setRrdatas(ImmutableList.copyOf(addition.getRrdatas()));
+          rrset.setTtl(addition.getTtl());
+          rrset.setType(addition.getType());
+          String id = getUniqueId(copy.keySet());
+          copy.put(id, rrset);
+        }
+      }
+      // make it immutable and replace
+      boolean success = dnsRecords.compareAndSet(original, ImmutableSortedMap.copyOf(copy));
+      if (success) {
+        break; // success if no other thread modified the value in the meantime
+      }
+    }
+    change.setStatus("done");
+  }
+
+  /**
+   * Lists zones. Next page token is the last listed zone name and is returned only of there is more
+   * to list.
+   */
+  @VisibleForTesting
+  Response listZones(String projectId, String query) {
+    Map<String, Object> options = OptionParsers.parseListZonesOptions(query);
+    Response response = checkListOptions(options);
+    if (response != null) {
+      return response;
+    }
+    ConcurrentSkipListMap<String, ZoneContainer> containers = findProject(projectId).zones();
+    String[] fields = (String[]) options.get("fields");
+    String dnsName = (String) options.get("dnsName");
+    String pageToken = (String) options.get("pageToken");
+    Integer maxResults = options.get("maxResults") == null
+        ? null : Integer.valueOf((String) options.get("maxResults"));
+    boolean sizeReached = false; // maximum result size was reached, we should not return more
+    boolean hasMorePages = false; // should next page token be included in the response?
+    LinkedList<String> serializedZones = new LinkedList<>();
+    String lastZoneName = null;
+    ConcurrentNavigableMap<String, ZoneContainer> fragment =
+        pageToken != null && containers.containsKey(pageToken)
```

Comment:
The service actually does use zone names as tokens. But I ran more tests and you are correct. You can even use a token which is not in the collection of names and it will simply list all the names that are greater than the token. I must have incidentally run all the tests using non-existent names such as "asadfsd" which were always smaller than all my created "zone1" "zone2" etc.

I removed the `contains` checks so the behaviour should now be identical.